src/lib.rs

```diff
@@ -1102,8 +1129,11 @@ mod impl_zerocopy {
             let my_candidate =
                 unsafe { candidate.assume_validity::<zerocopy::pointer::invariant::Valid>() };
             {
-                (my_candidate.read_unaligned::<zerocopy::pointer::BecauseImmutable>() ^ T::ALL_BITS)
-                    == T::EMPTY
+                // TODO: Currently this assumes that the candidate is aligned. We actually need to check this beforehand
```

Comment:
Not sure if this is something I would've caught. Do you have any links I could reference for example zerocopy implementations for similar types?